ocp_resources/datavolume.py

```diff
@@ -308,8 +310,10 @@ def test_dv():
                 wait_timeout=timeout,
                 func=lambda: self.exists,
             ):
-                # DV reach to success if the status is succeeded or if the DV does not exist
-                if sample is None or sample.get("status", {}).get("phase") == self.Status.SUCCEEDED:
+                # DV reach success if the status is Succeeded, or if DV garbage collection enabled and the DV does not exist
+                if (dv_garbage_collection_enabled and sample is None) or (
```

Comment:
with this code, we force wait for the DV to be gone if `dv_garbage_collection_enabled=True`
Is this the right thing to do here?

Reply:
Yes, it's the right thing to do

Reply:
When DV garbage collection is enabled, DV will reach the Succeeded Phase and then will be gone. Only PVC will stay.

Reply:
If garbage collection has a bug - this function **must** fail, because DV behaves not as expected and we can't be sure that DV succeeded. It would require further analysis - whether it's just DV wasn't deleted, or something is preventing import completion.

Reply:
No, PVC is created and working, this function shouldn't fail.
This is not test function.

Reply:
In the correct flow with DV garbage collection enabled - DV must be gone as soon as the import/upload/clone succeeded, we may not even catch thee DV's Succeeded phase. If DV is not gone - something is wrong, we don't know if the import finished or not, and we don't know if PVC is really working and contains the whole image.